src/elm/Pages/Build/Logs.elm

```diff
@@ -342,27 +349,91 @@ downloadFileName org repo buildNumber resourceType resourceNumber =
     String.join "-" [ org, repo, buildNumber, resourceType, resourceNumber ]
 
 
+{-| processLog : takes a log as string, splits it by newline, and processes it into a model that can render custom elements like timestamps and links
+-}
+processLog : String -> List (LogLine msg)
+processLog log =
+    log
+        |> String.split "\n"
+        |> List.map
+            (\log_ ->
+                let
+                    -- first we convert the log into an ANSI Log Model
+                    -- this lets us preserve the original ANSI style while applying our own processing
+                    ansiLogModel =
+                        Ansi.Log.update log_ defaultANSILogModel
+
+                    -- next we take the ANSI Model, run custom processing like link parsing, then render it to Html
+                    logView =
+                        -- we need to ignore extra empty lines generated by Ansi.Log.update
+                        case List.head <| Array.toList ansiLogModel.lines of
+                            Just logLine ->
+                                -- we may need to extend this to accept user preferences
+                                -- for example, a user may choose to not render links
+                                processAndViewLine logLine
+
+                            Nothing ->
+                                text ""
```

Comment:
i expect this section of code to get refactored when we introduce of log timestamps in https://github.com/go-vela/community/issues/243

this is where we would parse metadata and check the user model then do custom rendering.